amazonec2/amazonec2_endpoint.bal

```diff
@@ -177,6 +182,10 @@ public remote function Client.runInstances(string imgId, int maxCount, int minCo
     canonicalQueryString = canonicalQueryString + "Version" + "=" + API_VERSION;
     string constructCanonicalString = "/?" + canonicalQueryString;
     request.setHeader(HOST, host);
+    string token = self.securityToken;
```

Comment:
There seems to be significant code duplication here. Can you check if we can clean it up please?